index.js

```diff
@@ -119,16 +99,80 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:((?!\.|_)@?[0-9a-zA-Z./_%-]{1,214}):)?([^]+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (Nullable String, String, Number)
+  //                      -> TypeIdentifier
+  function TypeIdentifier(namespace, name, version) {
+    return {
+      namespace: namespace,
+      name: name,
+      version: version
+    };
+  }
+
+  //       parseTypeIdentifier :: String -> TypeIdentifier
+  function parseTypeIdentifier(s) {
+    if (s === '') return TypeIdentifier(null, s, 1);
+    var parsed = RPARSE.exec(s);
+    return TypeIdentifier(
+      parsed[1] == null ? null : parsed[1],
+      parsed[2],
+      parsed[3] == null ? 1 : parseInt(parsed[3], 10)
```

Comment:
Hm, they both have their quirks:

```js
Number('1a') = NaN
parseInt('1a', 10) = 1

parseInt('0xff', 10) = NaN
Number('0xff') = 255
parseInt('0xff') = 255

parseInt('1.1') = 1
Number('1.1') = 1.1
```

So whichever we use, we still have to verify the output, or restrict the input.